sthgetter/sthgetter.go

```diff
@@ -7,27 +7,30 @@ import (
 	"log"
 	"time"
 
+	ct "github.com/google/certificate-transparency-go"
 	"github.com/google/certificate-transparency-monitor/client"
+	"github.com/google/certificate-transparency-monitor/monitor"
+	"github.com/google/certificate-transparency-monitor/storage"
 )
 
 var logStr = "STH Getter"
 
 // Run runs an STH Getter, which periodically gets an STH from a Log, checks
 // that each one meets per-STH requirements defined in RFC 6962, and stores
 // them.
-func Run(ctx context.Context, lc *client.LogClient, url string, period time.Duration) {
+func Run(ctx context.Context, lc *client.LogClient, st storage.APICallWriter, url string, period time.Duration) {
 	log.Printf("%s: %s: Started with period %v", url, logStr, period)
 
 	t := time.NewTicker(period)
 	for {
 		select {
 		case <-t.C:
-			getCheckStoreSTH(url, lc)
+			getCheckStoreSTH(ctx, url, lc, st)
```

Comment:
I used to set timeouts at a lower level, for individual operations, which sounds a bit like what you're thinking, with timeouts possibly tailored to whatever the operation is (a Spanner read with a weak consistency ought to be done very quickly, but a write would have a longer timeout, for example)... But nowadays, my thinking is more toward a single high level timeout for the whole "request", and not fiddle with each sub-operation.

In particular, if you think about it, the question is this: is it ever okay for `getCheckStoreSTH` to never finish? Because that's what we're saying now, passing that indefinite context... Seems like something related to `period` might actually be what we want? After that, a sub-operation might use a tighter timeout, as well, but it wouldn't be strictly required.

Either way, I was only suggesting a TODO, this can be decided later, for this PR, go with this.

Reply:
TODO added :)